src/main/java/org/folio/service/loader/BlockingRecordLoaderService.java

```diff
@@ -1,51 +1,56 @@
+import io.vertx.core.json.JsonArray;
+import io.vertx.core.json.JsonObject;
+import org.folio.clients.impl.SourceRecordStorageClient;
+import org.folio.util.OkapiConnectionParams;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.Collection;
 import java.util.HashSet;
 import java.util.List;
 import java.util.Optional;
 import java.util.Set;
 
-import org.folio.clients.SynchronousOkapiClient;
-import org.springframework.stereotype.Service;
-
-import io.vertx.core.json.JsonArray;
-import io.vertx.core.json.JsonObject;
-
 /**
  * Synchronous implementation of #RecordLoaderService that uses blocking http client.
  */
 @Service
 public class BlockingRecordLoaderService implements RecordLoaderService {
 
-  private final SynchronousOkapiClient client;
-
-  public BlockingRecordLoaderService(SynchronousOkapiClient client) {
-    this.client = client;
-  }
+  @Autowired
+  private SourceRecordStorageClient client;
 
   @Override
-  public SrsLoadResult loadMarcRecords(List<String> uuids) {
-    Optional<JsonObject> optionalRecords = client.getByIds(uuids);
+  public SrsLoadResult loadMarcRecords(List<String> uuids, OkapiConnectionParams okapiConnectionParams) {
+    Optional<JsonObject> optionalRecords = client.getByIds(uuids, okapiConnectionParams);
     SrsLoadResult srsLoadResult = new SrsLoadResult();
     optionalRecords.ifPresent(records -> populateLoadResult(uuids, records, srsLoadResult));
     return srsLoadResult;
   }
 
-  private void populateLoadResult(List<String> queriedUuids, JsonObject foundRecords, SrsLoadResult loadResult) {
-    JsonArray records = foundRecords.getJsonArray("records");
-    Set<String> marcRecords = new HashSet<>();
-    Set<String> setSingleInstanceIdentifiers = new HashSet<>(queriedUuids);
+  @Override
+  public List<JsonObject> loadInventoryInstances(Collection<String> instanceIds, OkapiConnectionParams params) {
+    return new ArrayList<>();
+  }
+
+  private void populateLoadResult(List<String> uuids, JsonObject underlyingRecords, SrsLoadResult loadResult) {
+    JsonArray records = underlyingRecords.getJsonArray("records");
+    List<String> marcRecords = new ArrayList<>();
+    Set<String> singleInstanceIdentifiersSet = new HashSet<>(uuids);
     for (Object o : records) {
       JsonObject record = (JsonObject) o;
       marcRecords.add(getRecordContent(record));
       JsonObject externalIdsHolder = record.getJsonObject("externalIdsHolder");
       if (externalIdsHolder != null) {
         String instanceId = externalIdsHolder.getString("instanceId");
-        setSingleInstanceIdentifiers.remove(instanceId);
+        singleInstanceIdentifiersSet.remove(instanceId);
       }
     }
     loadResult.setUnderlyingMarcRecords(marcRecords);
-    loadResult.setSingleInstanceIdentifiers(setSingleInstanceIdentifiers);
+    loadResult.setInstanceIdsWithoutSrs(new ArrayList<>(singleInstanceIdentifiersSet));
```

Comment:
Can we use arrayList instead of set to avoid that wrapping? (I know that set remained from previous implementation)